lib/internal/url.js

```diff
@@ -1406,14 +1404,21 @@ function getPathFromURL(path) {
 // as this is the only character that won't be percent encoded by
 // default URL percent encoding when pathname is set.
 const percentRegEx = /%/g;
-function getURLFromFilePath(filepath) {
+function pathToFileURL(filepath) {
   const tmp = new URL('file://');
   if (filepath.includes('%'))
     filepath = filepath.replace(percentRegEx, '%25');
   tmp.pathname = filepath;
   return tmp;
 }
 
+function toPathIfFileURL(fileURLOrPath) {
+  if (fileURLOrPath == null || !fileURLOrPath[searchParams] ||
+      !fileURLOrPath[searchParams][searchParams])
+    return fileURLOrPath;
```

Comment:
What's the `fileURLOrPath[searchParams][searchParams]` check about? 
The `fileURLOrPath[searchParams]` checks currently allow paths to duck-type as URLs.

Reply:
This is discussed at https://github.com/nodejs/node/pull/22506#pullrequestreview-149374507.

Reply:
Specifically asking about the double `searchParams` check of 
`fileURLOrPath[searchParams][searchParams]`.

Reply:
This is exactly what is already implemented. We can switch to an `instanceof`, but would be good to hear from @jasnell if this would be compatible with the original logic.

Reply:
I would prefer a brand check (i.e. `[searchParams]`) as it's already been used everywhere else where we are checking for a `URL`, for consistency if nothing else. The same logic would probably apply to `fileURLToPath()` as well.

Reply:
Duck-typing isn't great. That it slipped in elsewhere in the project isn't a reason to continue it. It should really be reworked out when possible.

Reply:
I managed to trace down the origin of this code, and it was a performance PR made here - https://github.com/nodejs/node/pull/11690.

If we want to change this, we should do it in all of URL and not just this function, and as a separate PR.

Reply:
a note, we shouldn't call forgable things "branding" generally. "tagging" is generally the term here (like Symbol.toString**Tag**)